RIOT-OS/RIOT
# tests: rework eth drivers

### Contribution description
Split from #17739, but with a difference: instead of using symlinks the common functionality to interact with Ethernet netdev drivers is implemented in its own test_utils module.

### Testing procedure
- Run the tests, you should be able to send an receive Ethernet frames
- Green CI

### Issues/PRs references
#17739
